0017-ilp-errors/0017-ilp-errors.md

```diff
@@ -0,0 +1,73 @@
+# Interledger Protocol Errors
+
+This spec defines a standard error format and codes to be used when ILP payments are rejected.
+
+Inspired by [HTTP Status Codes](https://tools.ietf.org/html/rfc2616#section-10), ILP errors are categorized based on the intended behavior of the caller when they get the given error.
+
+## Error Propagation
+
+ILP connectors MUST relay errors when they are notified of a transfer being rejected.
+
+Connectors SHOULD NOT modify the error codes or messages except in the following way: connectors MAY append their ILP addresses, separated from the error message by a newline (`\n`), to assist with debugging/tracing.
+
+## Data Format
+
+ILP errors are UTF-8 encoded strings that are a maximum of 65535 bytes long. Error messages that are too long for a ledger to handle SHOULD be truncated to the first 65535 bytes.
+
+Errors consist of a three character code, followed by a colon (`:`) and an arbitrary message set by the entity that raised the error. Transport and application layer protocols MAY define error messages that have special significance in their context.
```

Comment:
Sorry, to clarify, these are not actual HTTP errors. My suggestion was to define ILP errors using the HTTP header and data format (in the same way we used it for [PSK](https://github.com/interledger/rfcs/blob/master/0016-pre-shared-key/0016-pre-shared-key.md#data-format)).

Errors would be propagated through ILP connectors (even if they are not using HTTP to talk to one another) via the "rejection message", which a sender or connector would get back instead of the ILP condition fulfillment.